asgs_main.sh

```diff
@@ -726,32 +726,96 @@ prepFile()
+   echo "hpc.slurm.job.${JOBTYPE}.reservation : ${_RESERVATION}" >> $STORMDIR/run.properties
+   echo "hpc.slurm.job.${JOBTYPE}.constraint : $CONSTRAINT" >> $STORMDIR/run.properties
+   echo "hpc.slurm.job.${JOBTYPE}.qos : $QOS" >> $STORMDIR/run.properties
    #
    # start log redirect processes for centralized logging
    initCentralizedScenarioLogging
    #
+   # build queue script
+   qScriptRequestTemplate=$SCRIPTDIR/qscript_request_template.json
+   qScriptRequest=$SCENARIODIR/qscript_request_$JOBTYPE.json
+   qScriptResponse=$SCENARIODIR/qscript_response_$JOBTYPE.json
+   QSCRIPTTEMPLATE=$SCRIPTDIR/qscript.template
+   parallelism=serial
+   forncpu=$NCPU
+   DATETIME=$(date +'%Y-%h-%d-T%H:%M:%S%z')
+   # keep sed from getting confused by escaping slashes
+   escSCRIPTDIR=${SCRIPTDIR////'\/'}
+   escADCIRCDIR=${ADCIRCDIR////'\/'}
+   escQSCRIPTTEMPLATE=${QSCRIPTTEMPLATE////'\/'}
+   escADVISDIR=${ADVISDIR////'\/'}
+   escSYSLOG=${SYSLOG////'\/'}
+   escSCENARIOLOG=${SCENARIOLOG////'\/'}
+   #
+   # create queue script request by filling in template
+   # with data needed to create queue script
+   sed \
+      -e "s/%jobtype%/$JOBTYPE/" \
+      -e "s/%qscripttemplate%/$escQSCRIPTTEMPLATE/" \
+      -e "s/%parallelism%/$parallelism/" \
+      -e "s/%ncpu%/$NCPU/" \
+      -e "s/%forncpu%/$NCPU/" \
+      -e "s/%numwriters%/$NUMWRITERS/" \
+      -e "s/%joblauncher%/$JOBLAUNCHER/" \
+      -e "s/%walltime%/$ADCPREPWALLTIME/" \
+      -e "s/%walltimeformat%/$WALLTIMEFORMAT/" \
+      -e "s/%ppn%/${_PPN}/" \
+      -e "s/%queuename%/$QUEUENAME/" \
+      -e "s/%serqueue%/$SERQUEUE/" \
+      -e "s/%account%/$ACCOUNT/" \
+      -e "s/%advisdir%/$escADVISDIR/" \
+      -e "s/%scriptdir%/$escSCRIPTDIR/" \
+      -e "s/%adcircdir%/$escADCIRCDIR/" \
+      -e "s/%scenario%/$SCENARIO/" \
+      -e "s/%reservation%/${_RESERVATION}/" \
+      -e "s/%constraint%/$CONSTRAINT/" \
+      -e "s/%qos%/$QOS/" \
+      -e "s/%syslog%/$escSYSLOG/" \
+      -e "s/%scenariolog%/$escSCENARIOLOG/" \
+      -e "s/%hotstartcomp%/$HOTSTARTCOMP/" \
+      -e "s/%queuesys%/$QUEUESYS/" \
+      -e "s/%hpcenvshort%/$HPCENVSHORT/" \
+      -e "s/%asgsadmin%/$ASGSADMIN/" \
+      -e "s/%NULLLASTUPDATER%/$THIS/" \
+      -e "s/%NULLLASTUPDATETIME%/$DATETIME/" \
+      < $qScriptRequestTemplate \
+      > $qScriptRequest \
+    2>> $SYSLOG
+   unset _PPN
+   unset _RESERVATION
+   # generate queue script
+   $SCRIPTDIR/qscript.pl < $qScriptRequest   \
+                         > $qScriptResponse 2>> $SYSLOG
+   if [[ $? != 0 ]]; then
+      fatal "Failed to generate queue script."
+   fi
+   # extract queue script name from response
+   qscript=$(bashJSON.pl --key "qScriptFileName"        \
+                        < $qScriptResponse 2>> $SYSLOG)
+   # extract queue script from response
+   bashJSON.pl --key "script" < $qScriptResponse 2>> $SYSLOG \
+                              | base64 -d                    \
```

Comment:
`base64` should be okay here, it is in GNU coreutils (had to look it up) - made me think to say, we just need to make sure any utils we use have a near 100% likelihood of being present. Stuff like `sed`, `awk`, etc are easy to assume.  Anyway, just wanted to point that out.